sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -1103,7 +1103,9 @@ object SparkSession extends Logging {
         loadExtensions(extensions)
         applyExtensions(sparkContext, extensions)
 
-        session = new SparkSession(sparkContext, None, None, extensions, options.toMap)
+        // Some config might supplemented. For example: spark.driver.extraJavaOptions.
+        val supplementedOptions = options.map { case (k, _) => k -> sparkContext.conf.get(k) }
```

Comment:
Does `sparkContext.conf` contain all entries in options ? What if there is an active SparkContext before creating SparkSession.

Reply:
Makes it exclude configs starting with `SPARK_DRIVER_PREFIX` and `SPARK_EXECUTOR_PREFIX`.